x/ratelimit/client/cli/cli_test.go

```diff
@@ -0,0 +1,318 @@
+		Outflow:      0,
+		ChannelValue: s.cfg.StakingTokens.Uint64(),
+	}
+
+	testCases := []struct {
+		name    string
+		args    []string
+		expFlow types.Flow
+	}{
+		{
+			"reset-rate-limit tx",
+			[]string{
+				sdk.DefaultBondDenom, // denom
+				path.ChannelId,       // channelId
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+				fmt.Sprintf("--%s=%s", flags.FlagFrom, val.Address.String()),
+				// common args
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				strideclitestutil.DefaultFeeString(s.cfg),
+			},
+			flow,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		s.Run(tc.name, func() {
+			cmd := cli.CmdAddRateLimit()
+			clientCtx := val.ClientCtx
+
+			// Add a new ratelimit
+			_, err := clitestutil.ExecTestCLICmd(clientCtx, cmd, []string{
+				sdk.DefaultBondDenom,                    // denom
+				path.ChannelId,                          // channelId
+				fmt.Sprintf("%d", quota.MaxPercentSend), // maxPercentSend
+				fmt.Sprintf("%d", quota.MaxPercentRecv), // maxPercentRecv
+				fmt.Sprintf("%d", quota.DurationHours),  // durationHours
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+				fmt.Sprintf("--%s=%s", flags.FlagFrom, val.Address.String()),
+				// common args
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				strideclitestutil.DefaultFeeString(s.cfg),
+			})
+			s.Require().NoError(err)
+
+			// Reset the rate limit
+			cmd = cli.CmdResetRateLimit()
+			_, err = clitestutil.ExecTestCLICmd(clientCtx, cmd, tc.args)
+			s.Require().NoError(err)
+
+			// Check if ratelimit was reset properly
+			cmd = cli.GetCmdQueryRateLimit()
+			out, err := clitestutil.ExecTestCLICmd(clientCtx, cmd, []string{
+				path.Denom,
+				path.ChannelId,
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+			})
+			s.Require().NoError(err)
+
+			var result types.RateLimit
+			s.Require().NoError(clientCtx.Codec.UnmarshalJSON(out.Bytes(), &result))
+			s.Require().Equal(tc.expFlow, *result.Flow)
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestCmdTxRemoveRateLimit() {
+	val := s.network.Validators[0]
+
+	rateLimit := types.RateLimit{}
+	denom := sdk.DefaultBondDenom
+	channelId := "channel-0"
+
+	testCases := []struct {
+		name         string
+		args         []string
+		expRateLimit types.RateLimit
+	}{
+		{
+			"remove-ratelimit tx",
+			[]string{
+				denom,     // denom
+				channelId, // channel id
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+				fmt.Sprintf("--%s=%s", flags.FlagFrom, val.Address.String()),
+				// common args
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				strideclitestutil.DefaultFeeString(s.cfg),
+			},
+			rateLimit,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		s.Run(tc.name, func() {
+			cmd := cli.CmdRemoveRateLimit()
```

Comment:
I think we should add a rate limit first before testing the remove, something like:
1. Attempt to remove a rate limit (should fail)
2. Add a rate limit
3. Remove that rate limit (should succeed)
4. Check that rate limit was actually removed

Might also make sense to bundle the add rate limit into its own function that you can call from this and ResetRateLimit

Reply:
Ok, actually it doesn't fail as there are already rate limits added while running previous tests but I will try to add a new one for running remove cli.
And, we already have a very similar test scenario(1-4) on our simapp tests so I don't think to repeat that again for cli tests.

Reply:
> Ok, actually it doesn't fail as there are already rate limits added while running previous tests but I will try to add a new one for running remove cli.

Ah my mistake, I think I misunderstood how these CLI tests run

> And, we already have a very similar test scenario(1-4) on our simapp tests so I don't think to repeat that again for cli tests.

Totally agree on not duplicating that work again